### PR TITLE
Implement issue #734

### PR DESCRIPTION
**Pull request by KaiZin2018**
Show different Item icon in tree view for entries that will be shortly expired (warning) or are expired.

### CODE REVIEW

File: src/ui/wxWidgets/TreeCtrl.cpp [hunk 1]
```diff
@@ -838,14 +839,50 @@ bool TreeCtrl::Remove(const CUUID &uuid)
 void TreeCtrl::SetItemImage(const wxTreeItemId &node,
                                const CItemData &item)
 {
-  // TODO: modify to display warning and expired states
+  int offset = 0;
+  CItemData::EntryType entrytype = item.GetEntryType();
+  
+  if((entrytype == CItemData::ET_NORMAL) || (entrytype == CItemData::ET_ALIASBASE) || (entrytype == CItemData::ET_SHORTCUTBASE)) {
+    time_t tttXTime(static_cast<time_t>(0));
```
Comment (dskrvk) [inline src/ui/wxWidgets/TreeCtrl.cpp:846]: Minor: the method names GetXTime, GetPMTime, and GetCTime are not very intuitive and make the calculations here a bit hard to follow. Maybe we could at least name the corresponding local variables differently - e.g. `daysToExpiration` & `creationModificationTimestamp`?

File: src/ui/wxWidgets/TreeCtrl.cpp [hunk 2]
```diff
@@ -838,14 +839,50 @@ bool TreeCtrl::Remove(const CUUID &uuid)
...
+    item.GetXTime(tttXTime);
+    if (tttXTime > static_cast<time_t>(0) && tttXTime <= static_cast<time_t>(3650)) {
+      time_t tttCPMTime(static_cast<time_t>(0));
+      item.GetPMTime(tttCPMTime);
+      if (tttCPMTime == static_cast<time_t>(0))
+        item.GetCTime(tttCPMTime);
+      tttXTime = static_cast<time_t>((long)tttCPMTime + (long)tttXTime * 86400L);
```
Comment (dskrvk) [inline src/ui/wxWidgets/TreeCtrl.cpp:853]: What's happening on this line? Is it trying to correct an invalid tttXTime?

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 1 above)
Comment (KaiZin2018) [reply src/ui/wxWidgets/TreeCtrl.cpp:858]: I added comments for those who are not so familiar with the short form of field names.

File: src/ui/wxWidgets/TreeCtrl.cpp (hunk 2 above)
Comment (KaiZin2018) [reply src/ui/wxWidgets/TreeCtrl.cpp:868]: Add comment about handling of time versus handling of distance of days before expiry.

**Comment by KaiZin2018** [comment]
The change in TreeCtrl::Remove() had lead to left wrongly empty groups when undo tree Drag and Drop. Recursive deletion as done before would have removed empty groups when present before empty groups. With additional change the group structure in tree view is left correct when undo tree Drag and Drop action.

**Comment by ronys** [comment]
Nice work - thanks!
